src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs

```diff
@@ -1084,6 +1096,10 @@ private static XNode[] RewriteMany(ISymbol symbol, Compilation compilation, XNod
             {
                 return Array.Empty<XNode>();
             }
+            finally
+            {
+                visitedSymbols.Remove(symbol);
```

Comment:
Why is this needed?

Reply:
Symbols can appear in more than one reference in the same comment. They just can't be nested.

Reply:
This absolutely needs a comment....

Reply:
And to check there's an existing test that includes a symbol twice that triggers this?

Reply:
I'll get the comment and test added for 16.5. preview 1 branch